openedx/core/djangoapps/discussions/docs/decisions/0006-discussions-api-fetch-inactive-providers.rst

```diff
@@ -0,0 +1,82 @@
+Allow fetching configuration for inactive providers
+===================================================
+
+
+Status
+------
+
+Proposal
+
+
+Context
+-------
+
+In ADR 003 an API was proposed for discussion configuration. This proposed a
+single API that returned information about the active provide and listed other
+available providers.
+
+The API was designed to only return information about the current active
+provider however in practise this creates an issue when switching providers.
+
+The troublesome scenario is as follows:
+
+- A course author configures the legacy provider
+- They switch to another provider
+- Now they switch back to the legacy provider
+- They have to set up the legacy provider from scratch again
```

Comment:
I think in this scenrio, redux would have the legacy data available. 

Actually the issue was when the author has configured an LTI provider 
* Visits discussion settings
* Switches back to the legacy provider
* No settings/data available for them to set up. 2) if they add new data, the existing topic ids would be lost and removed and all associated discussion posts would also be stale.

Reply:
@awaisdar001 The above series of steps isn't meant to be a single session. i.e. after switching from the legacy provider to another provider, the tab is closed, the author tests out the provider, then wants to swich back. 
Redux will not have the data at that point. 

If a user is switching from a LTI to the legacy provider for the first time this isn't needed because in that case we need the default settings which we already have.